wallet/udb/vsp.go

```diff
@@ -26,15 +28,55 @@ const (
 	// paid, but not published.
 	VSPFeeProcessPaid
 	VSPFeeProcessErrored
+	// VSPFeeProcessConfirmed represents the state where the fee has been
+	// confirmed by the VSP.
+	VSPFeeProcessConfirmed
 )
 
 type VSPTicket struct {
 	FeeHash     chainhash.Hash
 	FeeTxStatus uint32
+	VSPHostID   uint32
+	Host        string
+	PubKey      []byte
 }
 
 // SetVSPTicket sets a VSPTicket record into the db.
 func SetVSPTicket(dbtx walletdb.ReadWriteTx, ticketHash *chainhash.Hash, record *VSPTicket) error {
+	// Check for Host/Pubkey buckets
+	pubkey, err := GetVSPPubKey(dbtx, []byte(record.Host))
+	if err != nil && errors.Is(err, errors.NotExist) {
```

Comment:
Re-reding, one limitation here is that this doesn't allow replacing the pubkey for the given host if it ever changes. 

Maybe after fetching the pubkey via `getVSPPubKey`, check if it is `!= nil` and `!= record.PubKey` and then rewrite the pubkey in the DB to use the new one (while keeping the host ID intact). This would future-proof if a VSP ever changes its pubkey for any reason.